src/Object/Api/Mastodon/Status.php

```diff
@@ -95,7 +95,7 @@ class Status extends BaseEntity
 	 * @param array   $item
 	 * @throws \Friendica\Network\HTTPException\InternalServerErrorException
 	 */
-	public function __construct(array $item, Account $account)
+	public function __construct(array $item, Account $account, array $count)
```

Comment:
I'm a little disappointed. When you asked whether you should group parameters in an associative array, you mentioned the awkward case where there would be "20 parameters". Instead, you used an array parameter to replace only three integer parameters, which obscures the code for no discernable benefit.

Please use individual parameters for these three counts.
```suggestion
	public function __construct(array $item, Account $account, int $replies_count, int $reblogs_count, int $favourites_count)
```

Reply:
Well. In the future there will be the need for parameters for:
* sensitive
* language
* replies_count
* reblogs_count
* favourites_count
* favourited
* reblogged
* muted
* bookmarked
* pinned
* reblog
* application
* media_attachments
* mentions
* tags
* emojis
* card
* poll

So I think that it does make sense using an array for the counts and a different array for the fields that mark the user interactions (favourited, reblogged, muted, bookmarked, pinned) since they are thematically grouped.

Reply:
The count array parameter still would "remove" only two parameters, which isn't really significant.

Reply:
But also there is the group of the 4 parameters I mentioned above. So it would be 2 instead of 7 parameters - and in the following entities there would be more.

Reply:
2 parameters instead of 7 isn't a great case for me except if the 2 were structured data. There's no functional difference between setting 3 array keys and passing the array instead of passing 3 variables, but the code is harder to read because some parameters are hidden in arbitrary array keys.

Reply:
well, that's inconsequent. Following this idea shouldn't we then provide each and every field as variable?

Reply:
No, if we have local structured data we should use it. In particular database rows should be provided as is, even if they are just arrays at the moment. But single scalar values like counts or dates should be provided individually.

And I believe you meant "inconvenient"?

Reply:
I meant the meaning of "Inconsistent". I understood this concept as "moving the whole business logic out of this class". And this can only be done in perfection by using a single parameter for each field.

Reply:
It isn't so much "moving the whole business logic out of this class" rather than "not having any hard dependency". Since it's transforming our data structure in the target data structure, there can be operators, array accesses and such, but we shouldn't need any database call in this class.

Reply:
I see these arrays as some tool to group data. We have two groups here: The "count" values and the user interaction. I can understand why you don't like the arrays. But what about classes?

What about this way:
````
$count = new stdClass;
$count->replies = DBA::count('item', ['thr-parent-id' => $uriId, 'uid' => $uid, 'gravity' => GRAVITY_COMMENT]);
$count->reblogs = DBA::count('item', ['thr-parent-id' => $uriId, 'uid' => $uid, 'gravity' => GRAVITY_ACTIVITY, 'vid' => Verb::getID(Activity::ANNOUNCE)]);
$count->favourites = DBA::count('item', ['thr-parent-id' => $uriId, 'uid' => $uid, 'gravity' => GRAVITY_ACTIVITY, 'vid' => Verb::getID(Activity::LIKE)]);
````

Or possibly we could define classes for these structures?

Reply:
If you create a class `Counts`, I will gladly accept a single parameter of the class `Counts`. `stdClass` is just anothe associative array with no structure information, so I will not accept it.

I want the code to be as explicit as possible, and a function array parameter must be a list of things, not an opaque associative array. We've had numerous issues and Notice messages because one of the keys we expected from an associative array wasn't set in some particular situation and I want to suppress them preventatively in new code.

Reply:
The `Counts` class should look like this:
```
namespace Friendica\Object\Api\Mastodon\Account;

class InteractionCounts
{
    protected $replies;
    protected $reblogs;
    protected $favourites;

    public function _construct(int $replies, int $reblogs, int $favourites) {
        $this->replies = $replies;
        $this->reblogs = $reblogs;
        $this->favourites = $favourites;
    }

    public function __get($name) {
        return $this->$name;
    }
}
```
This prevents property modification ensuring the queries used to generate the counts in the first place must be accurate.

Reply:
@MrPetovan this sounds like a good plan to me.

Reply:
Is it an account count?

Reply:
It is for the `status` entity.

Reply:
How can I now use this class?

Reply:
Sorry about my silence, in this case I believe `Status\Counts` would be suitable.